### PR TITLE
Feat: cart component

### PR DESCRIPTION
**Pull request by oknelvapi**
pr
What was done:
- created Cart component;
- created Shapes components;
- installed [react-if](https://www.npmjs.com/package/react-if)  package
- added [pull_request_template](https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository)

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+**Name**
```
Comment (braindrain9) [inline pull_request_template.md:1]: good call 👍  but I would make this change in a separate PR - these changes are not related to current PR.

File: src/shared/components/Cart/component.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { Oval, ParallelogramOvalToLeft, RectangleOvalToRight } from 'shared/shapes';
+import { CartWrapper, ImageWrapper, CartContent, CartSubHeader } from './styles';
+
+type CartProps = {
+  imgName?: string;
+  imgURL: string;
+  caption: string;
+  title: string;
+  text?: string;
+  size: 'small' | 'medium' | 'large';
```
Comment (braindrain9) [inline src/shared/components/Cart/component.tsx:14]: let's create `enum` for size values; 

```js
enum CartSize {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
}
...
size: CartSize;
```

File: src/shared/components/Cart/component.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  textColor?: string;
```
Comment (braindrain9) [inline src/shared/components/Cart/component.tsx:15]: should be `textColor?: Colors` - and let's pass Colors values

File: src/shared/components/Cart/styles.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
+import styled from 'styled-components';
+import { Media } from 'react-bootstrap';
+
+import Colors from 'shared/colors';
+import media from 'shared/media-types';
+
+type CartWrapperProps = {
+  size: 'small' | 'medium' | 'large';
```
Comment (braindrain9) [inline src/shared/components/Cart/styles.ts:8]: `size: CartSize`;

File: src/shared/components/Cart/styles.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+};
+type CartSubHeaderProps = {
+  textColor?: string;
```
Comment (braindrain9) [inline src/shared/components/Cart/styles.ts:11]: `textColor?: Colors;`

File: src/shared/components/Cart/styles.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+};
+
+export const CartWrapper = styled.div<CartWrapperProps>`
+  width: 100%;
+  display: flex;
+  flex-flow: row wrap;
+  padding-left: 20px;
+
+  div:first-child {
+    ${({ size }): string => (size === 'large' ? 'margin-right: 60px' : '')};
```
Comment (braindrain9) [inline src/shared/components/Cart/styles.ts:24]: use types for `size`

File: src/shared/shapes/types.d.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+type ShapeProps = {
```
Comment (braindrain9) [inline src/shared/shapes/types.d.ts:1]: doesn't it should be exported?

File: src/shared/components/Cart/styles.ts (hunk 6 above)
Comment (oknelvapi) [reply src/shared/components/Cart/styles.ts:24]: I used the 'generics' approach: 
```javascript
styled.div<CartWrapperProps>``;
``` 
Do you think it's a bad idea?

File: src/shared/components/Cart/styles.ts (hunk 5 above)
Comment (oknelvapi) [reply src/shared/components/Cart/styles.ts:11]: done

File: src/shared/components/Cart/styles.ts (hunk 4 above)
Comment (oknelvapi) [reply src/shared/components/Cart/styles.ts:8]: done

File: src/shared/components/Cart/component.tsx (hunk 2 above)
Comment (oknelvapi) [reply src/shared/components/Cart/component.tsx:14]: Done

File: src/shared/components/Cart/component.tsx (hunk 3 above)
Comment (oknelvapi) [reply src/shared/components/Cart/component.tsx:15]: Done

File: src/shared/shapes/types.d.ts (hunk 7 above)
Comment (oknelvapi) [reply src/shared/shapes/types.d.ts:1]: No, it does not. I created `d.ts` file. We don't have to export all definitions that were declared there. However, I have rewritten this file to simple `.ts` because I'm not sure that it is a good idea, hmm... We have to investigate the possibility to use a `d.ts`.

File: pull_request_template.md (hunk 1 above)
Comment (oknelvapi) [reply pull_request_template.md:1]: Ok, up to you.

File: src/shared/components/Cart/styles.ts (hunk 6 above)
Comment (braindrain9) [reply src/shared/components/Cart/styles.ts:22]: Yes, we should be strict here and use  defined `enums`, e.g. `size === ${Size.LARGE}`. We do not have to use 'string' values at all - as it can cause a lot of confusing and mix-ups in future.

Other benefit is it's easier to manage and if we update enum value, we can update it just in one place.

File: src/shared/shapes/Oval.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import { ShapeProps } from './types';
+
+const Oval: React.FC<ShapeProps> = ({ size, fill }: ShapeProps) => {
+  const getSize = (): number => {
+    if (size === 'large') return 246;
```
Comment (braindrain9) [inline src/shared/shapes/Oval.tsx:7]: please use enum here

File: src/shared/shapes/RectangleOvalToRight.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import { ShapeProps } from './types';
+
+const RectangleOvalToRight: React.FC<ShapeProps> = ({ size }: ShapeProps) => {
+  const getWidth = (): number => {
+    if (size === 'large') return 242;
```
Comment (braindrain9) [inline src/shared/shapes/RectangleOvalToRight.tsx:7]: the same here

File: src/shared/shapes/types.d.ts (hunk 7 above)
Comment (braindrain9) [reply src/shared/shapes/types.d.ts]: oh, I see. maybe, creating declaration files is a good idea and we should use it everywhere for other types. But we should check - on which scope these type values are exported (global or only for items in the same folder) and if these declaration files do not cause conflicts with other types which have the same name. (StyleProps etc.)

File: src/shared/colors.ts [hunk 10]
```diff
@@ -21,6 +21,8 @@ enum Colors {
   YELLOW_DARK = '#FFD167',
   YELLOW_LIGHT = '#FFECB8',
   WHITE = '#FFFFFF',
+  DARK_BACKGROUND = '#181818',
```
Comment (braindrain9) [inline src/shared/colors.ts:24]: it's already exist as `GREY_DARK`

File: src/shared/colors.ts [hunk 11]
```diff
@@ -21,6 +21,8 @@ enum Colors {
...
+  LIGHT_GREY_BACKGROUND = '#e5e5e5',
```
Comment (braindrain9) [inline src/shared/colors.ts:25]: we can use `GREY_ULTRA_MEDIUM_LIGHT` or `GREY_ULTRA_LIGHT`

File: src/shared/colors.ts (hunk 10 above)
Comment (oknelvapi) [reply src/shared/colors.ts:24]: I didn't use it and create it.  Button component uses it.

Comment (braindrain9) [reply src/shared/colors.ts:24]: one minor missed - please do not create the same variable here and use `GREY_DARK `

**Comment by oknelvapi** [comment]
> one minor missing

Please, look through the components. I didn't use it and I didn't create it. Git highlighted it because of my editing that file.  There is the 'button' component that uses it and that wasn't created by myself! Therefore, editing that button component isn't an issue of this PR.

**Comment by braindrain9** [comment]
@oknelvapi please be more attentive. Button component **does not** use this variable anymore (it was renamed to GREY_DARK). 

You should understand, that all changes in https://github.com/MecenatOrg/frontend/pull/23/files - **are actually the changes you request to merge to master**. So, you are just adding **unnecessary** and unused change.

I suppose, it is a leftover after merging conflicts with master (you had conflicts in this file and just resolved them not correctly).

So, please, just remove this variable.
